### PR TITLE
all: fetch testrun metadata from squad

### PR DESCRIPTION
**Pull request by jscook2345**
Treat the testrun metadata like a squadobject, and use the built-in
features to send and recieve it.

Originally this was treated like a dictionary, and so there are some
changes related to that.

Because the testrun metadata does not really use an id, we remove the
uuid when we encode it to json.

Signed-off-by: Justin Cook <justin.cook@linaro.org>

### CODE REVIEW

File: tests/test_models.py [hunk 1]
```diff
@@ -153,6 +153,11 @@ def setUp(self):
     def test_basic(self):
         self.assertTrue(self.testrun is not None)
 
+    def test_testrun_metadata(self):
+        self.assertTrue(self.testrun.metadata_file is not None)
+        self.assertTrue(self.testrun.metadata is not None)
+        self.assertEqual(self.testrun.metadata.foo, "bar")
```
Comment (chaws) [inline tests/test_models.py:159]: Nice!

**Comment by chaws** [comment]
LGTM, could you please just add a test covering a TestRun with None as metadata_file? I think in that case the `testrun.metadata` should return None as well. What do you think?

**Comment by jscook2345** [comment]
@chaws Done!
